code/game/machinery/suit_storage_unit.dm

```diff
@@ -261,47 +271,52 @@
 		uv_cycles = initial(uv_cycles)
 		uv = FALSE
 		locked = FALSE
-		if(uv_super)
-			visible_message("<span class='warning'>[src]'s door creaks open with a loud whining noise. A cloud of foul black smoke escapes from its chamber.</span>")
-			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 50, TRUE)
-			helmet = null
-			qdel(helmet)
-			suit = null
-			qdel(suit) // Delete everything but the occupant.
-			mask = null
-			qdel(mask)
-			storage = null
-			qdel(storage)
-			// The wires get damaged too.
-			wires.cut_all()
-		else
-			if(!occupant)
-				visible_message("<span class='notice'>[src]'s door slides open. The glowing yellow lights dim to a gentle green.</span>")
+		if(uv_super | (obj_flags & EMAGGED)) //NSV13 start of edited segment, adds Emag check, different occupant message if it's also hacked and makes the items get damaged instead of deleted.
+			toasted = TRUE
+			if(occupant)
+				visible_message("<span class='warning'>[src]'s door creaks open with a loud whining noise. A foul stench and a cloud of smoke exit the chamber.</span>")
 			else
-				visible_message("<span class='warning'>[src]'s door slides open, barraging you with the nauseating smell of charred flesh.</span>")
-				mob_occupant.radiation = 0
-			playsound(src, 'sound/machines/airlockclose.ogg', 25, TRUE)
-			var/list/things_to_clear = list() //Done this way since using GetAllContents on the SSU itself would include circuitry and such.
-			if(suit)
-				things_to_clear += suit
-				things_to_clear += suit.GetAllContents()
+				visible_message("<span class='warning'>[src]'s door creaks open with a loud whining noise. A cloud of foul black smoke escapes from its chamber.</span>")
+			playsound(src, 'sound/machines/airlock_alien_prying.ogg', 50, TRUE)
 			if(helmet)
-				things_to_clear += helmet
-				things_to_clear += helmet.GetAllContents()
+				helmet.take_damage(100,BURN,"fire")
+			if(suit)
+				suit.take_damage(100,BURN,"fire")
 			if(mask)
-				things_to_clear += mask
-				things_to_clear += mask.GetAllContents()
+				mask.take_damage(100,BURN,"fire")
 			if(storage)
-				things_to_clear += storage
-				things_to_clear += storage.GetAllContents()
+				storage.take_damage(100,BURN,"fire")
+			// The wires get damaged too.
+			wires.cut_all()
+		if(!toasted) //NSV13 Special toast check to prevent a double finishing message.
 			if(occupant)
-				things_to_clear += occupant
-				things_to_clear += occupant.GetAllContents()
-			for(var/atom/movable/AM in things_to_clear) //Scorches away blood and forensic evidence, although the SSU itself is unaffected
-				SEND_SIGNAL(AM, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_STRONG)
-				var/datum/component/radioactive/contamination = AM.GetComponent(/datum/component/radioactive)
-				if(contamination)
-					qdel(contamination)
+				visible_message("<span class='warning'>[src]'s door slides open, barraging you with the nauseating smell of charred flesh.</span>")
+				mob_occupant.radiation = 0
```

Comment:
Uh you want the mob to not be iiradiated but burned badly instead for being killed in it?

Reply:
This is already part of the original code for the SSU, it removes radioactive contamination from the occupant like it does with clothing because they just got cleaned VERY EFFECIENTLY.